### PR TITLE
[13.0][MIG] base_exception: Migration to 13.0

### PR DESCRIPTION
**Pull request by joao-p-marques**
Superseeds https://github.com/OCA/server-tools/pull/1760
Ran pre-commit and squashed latest commits that attended comments in previous PR.

@Tecnativa
TT21615

ping @pedrobaeza

### CODE REVIEW

**Comment by joao-p-marques** [comment]
Also updated README

**Review by atchuthan — APPROVED** [review]
Code review :+1:

File: base_exception/tests/purchase_test.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class LineTest(models.Model):
+    _name = "base.exception.test.purchase.line"
+    _description = "Base Exception Test Model Line"
+
+    name = fields.Char()
+    lead_id = fields.Many2one("base.exception.test.purchase", ondelete="cascade")
+    qty = fields.Float()
+    amount = fields.Float()
```
Comment (victoralmau) [inline base_exception/tests/purchase_test.py:69]: It's a good idea load new model to use it later in tests?

**Review by victoralmau — APPROVED** [review]

File: base_exception/tests/purchase_test.py (hunk 1 above)
Comment (joao-p-marques) [reply base_exception/tests/purchase_test.py:69]: Thanks for the comment @victoralmau 
Seems like a strange approach, indeed, but I have done some digging and it seems to be the best option to test this according to [this comment](https://github.com/OCA/server-tools/pull/1025#discussion_r153815296)
It already comes from previous versions...
@pedrobaeza do you think this should be refactored?

Comment (pedrobaeza) [reply base_exception/tests/purchase_test.py:69]: Let's keep it for now.

**Pull request merged**